### PR TITLE
CONSOLE-3910: Add quickstart to impersonate a system:admin user

### PR DESCRIPTION
**Pull request by sg00dwin**
Screenshot 2024-02-09 at 1 19 07 PM

Screenshot 2024-02-09 at 1 20 45 PM

Screenshot 2024-02-09 at 1 21 34 PM
Screenshot 2024-02-09 at 1 21 05 PM

### CODE REVIEW

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Comment by sg00dwin** [comment]
QE Approver
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

**Comment by RickJWagner** [comment]
Thank you for the screen shots and the nice, clean documentation @sg00dwin

**Comment by opayne1** [comment]
Thanks for looping me in!

File: quickstarts/impersonate-user.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  introduction: >-
+    You can grant a user cluster administrator permissions by giving them permission to impersonate `system:admin`. For more information, see [User impersonation](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation).
+  tasks:
+    - title:  Select the system:admin user to impersonate
+      description: >
+        You must explicitly have the `impersonate` privilege to impersonate other users:
+
+        1. From the **Administrator** perspective, go to the **RoleBindings** page in the [User Management]{{highlight qs-nav-usermanagement}} section of the navigation.
+
+        1. Sort by **Subject Kind** `User` and locate the **Subject Name** `system:admin` user you want to impersonate. Then click its associated kebab menu and select **Impersonate User "system:admin"**.
```
Comment (yapei) [inline quickstarts/impersonate-user.yaml:31]: The table column name is `Subject kind` and `Subject name` 
Shall we update to be consistent?

Comment (yapei) [reply quickstarts/impersonate-user.yaml:31]: Screenshot 2024-02-20 at 12 47 55 PM

**Comment by yapei** [comment]
in my local testing, clicking `Impersonate User xxx` or `Stop Impersonation` will close the quickstart modal, it doesn't seem expected
@sg00dwin could you help confirm?

File: quickstarts/impersonate-user.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          <p>"Impersonating User<br />
+          You are impersonating <b>system:admin</b>. You are viewing all resources and roles this User can access. Stop impersonation"</p>
+      summary:
+        failed: Try the steps again.
+        success: You have successfully impersonated a user.
+    - title:  Stop impersonation
+      description: >
+        Stop impersonating the system:admin user:
+
+        1.  Click the **Stop Impersonation** link in the notification.
```
Comment (yapei) [inline quickstarts/impersonate-user.yaml:47]: The button name is `Stop impersonation`

Comment (yapei) [reply quickstarts/impersonate-user.yaml:47]: Screenshot 2024-02-20 at 1 03 53 PM

**Comment by yapei** [comment]
@sg00dwin I see the quick start text is updated correctly, how about [this issue](https://github.com/openshift/console-operator/pull/865#issuecomment-1953489021)?

**Comment by jhadvig** [comment]
@yapei the issue should be fixed in the master, so the PR is ready for QE testing :)

**Comment by yapei** [comment]
@jhadvig  @sg00dwin The issue still exists, every time we click on `Impersonate User xxx` or `Stop impersonate` the quickstart progress will be reset and we have to start over again and again 

https://github.com/openshift/console-operator/assets/12692381/ea0ab7a6-0910-4204-88a3-ee1b1600b4e9

**Comment by sg00dwin** [comment]
> The issue still exists, every time we click on `Impersonate User xxx` or `Stop impersonate` the quickstart progress will be reset and we have to start over again and again
> 
Hi @yapei I was out on pto.  There was another issue blocking the work on this story, but a fix has been merged. 

The issue you mentioned still needs to be addressed.

**Comment by jhadvig** [comment]
@yapei we discussed this issues with @alimobrem today and we came to a conclusion that its actually by design. @sg00dwin will update the quickstart so it will notify the user that once he starts with impersonating, the QS will get hidden.

**Comment by yapei** [comment]
thanks for confirmation everyone ~
/label qe-approved

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

**Pull request merged**